server/sonar-server/src/test/java/org/sonar/server/edition/CommitPendingEditionOnStartupTest.java

```diff
@@ -0,0 +1,181 @@
+    when(editionManagementState.getPendingInstallationStatus()).thenReturn(MANUAL_IN_PROGRESS);
+
+    underTest.start();
+
+    verify(editionManagementState).getPendingInstallationStatus();
+    verifyNoMoreInteractions(editionManagementState);
+    verifyZeroInteractions(licenseCommit);
+    assertThat(logTester.logs()).hasSize(1);
+    assertThat(logTester.logs(LoggerLevel.DEBUG))
+      .containsOnly("No LicenseCommit instance is not available, can not finalize installation");
+  }
+
+  @Test
+  public void start_commit_license_and_finalizeInstallation_in_editionManagementState_when_status_is_MANUAL_IN_PROGRESS_and_LicenseCommit_is_available() {
+    when(editionManagementState.getPendingInstallationStatus()).thenReturn(MANUAL_IN_PROGRESS);
+    String license = RandomStringUtils.randomAlphanumeric(20);
+    when(editionManagementState.getPendingLicense()).thenReturn(Optional.of(license));
+
+    underTestWithLicenseCommit.start();
+
+    verify(editionManagementState).getPendingInstallationStatus();
+    verify(editionManagementState).getPendingLicense();
+    verify(editionManagementState).finalizeInstallation();
+    verifyNoMoreInteractions(editionManagementState);
+    verify(licenseCommit).update(license);
+    verifyNoMoreInteractions(licenseCommit);
+  }
+
+  @Test
+  public void starts_put_editionManagement_set_in_automaticInstallError_when_status_is_AUTOMATIC_PROGRESS_and_no_LicenseCommit_is_available() {
+    when(editionManagementState.getPendingInstallationStatus()).thenReturn(AUTOMATIC_IN_PROGRESS);
+
+    underTest.start();
+
+    verify(editionManagementState).getPendingInstallationStatus();
+    verify(editionManagementState).installFailed("SonarQube was restarted before asynchronous installation of edition completed");
+    verifyNoMoreInteractions(editionManagementState);
+    verifyZeroInteractions(licenseCommit);
+  }
+
+  @Test
+  public void starts_put_editionManagement_set_in_automaticInstallError_when_status_is_AUTOMATIC_PROGRESS_and_LicenseCommit_is_available() {
+    when(editionManagementState.getPendingInstallationStatus()).thenReturn(AUTOMATIC_IN_PROGRESS);
+
+    underTestWithLicenseCommit.start();
+
+    verify(editionManagementState).getPendingInstallationStatus();
+    verify(editionManagementState).installFailed("SonarQube was restarted before asynchronous installation of edition completed");
+    verifyNoMoreInteractions(editionManagementState);
+    verifyZeroInteractions(licenseCommit);
+  }
+
+  @Test
+  public void should_commit_uninstall() {
+    when(editionManagementState.getPendingInstallationStatus()).thenReturn(EditionManagementState.PendingStatus.UNINSTALL_IN_PROGRESS);
+
+    underTest.start();
+
+    verify(editionManagementState).getPendingInstallationStatus();
+    verify(editionManagementState).finalizeInstallation();
+    verifyNoMoreInteractions(editionManagementState);
+  }
+
+  @Test
+  @Ignore
+  public void should_fail_uninstall_if_license_commit_is_present() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1607)